### PR TITLE
Commit for Moka-001 , Moka-002 , Moka-003, Moka-004 

### PR DESCRIPTION
**Pull request by nazmy**
Hi Grady,

I have enhanced the previously submitted PR. Devise authentication has been implemented and cancancan authorisation has been improved in this version. I intentionally created three roles to show the usage of cancancan to define different privileges. 

I also have improved the controller test to test the privileges define in cancancan abilities. E.g user shall not be able to access, edit, delete resource (item & business) that are not belong to him. 

For your perusal please. I'd be grateful to have this opportunity to learn from the rails expert 😄. 

PS: User and business relation can be improved further by changing it to many-to-many since user may be assigned different roles in different business. However, I suppose this is not required for this interview.

Thanks,
Nazmy

### CODE REVIEW

**Comment by glaksmono** [comment]
https://screencast.com/t/RQgomRVSioC4

How do you do CRUD on Item?

Did you implement this part? https://github.com/glaksmono/moka-rails-interview/blob/02b427d336e10e78bf5ca72f97ba19df1c407d9c/app/views/items/index.html.erb#L9

File: app/controllers/businesses_controller.rb [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+	def create
+		if @business.save
+			redirect_to items_path, :notice => "Business '#{@business.name}' has been created successfully!"
+		else
+			render :new, :alert => "Unable to create business '#{@business.name}'. Error : '#{@business.errors.full_messages}'"
+		end
+	end
+
+	def update
+		if Business.where(:id => params[:id]).present?
```
Comment (glaksmono) [inline app/controllers/businesses_controller.rb:19]: Multiple indentation style

**Review by glaksmono — CHANGES_REQUESTED** [review]

**Comment by glaksmono** [comment]
[MOKA-001] Finish the implementation of Create/Update/Delete for Items
- Create: https://github.com/glaksmono/moka-rails-interview/blob/02b427d336e10e78bf5ca72f97ba19df1c407d9c/app/views/items/index.html.erb#L9
- Update: https://github.com/glaksmono/moka-rails-interview/blob/02b427d336e10e78bf5ca72f97ba19df1c407d9c/app/views/items/index.html.erb#L17
- Delete: https://github.com/glaksmono/moka-rails-interview/blob/02b427d336e10e78bf5ca72f97ba19df1c407d9c/app/views/items/index.html.erb#L18

**Comment by nazmy** [comment]
Hi Grady,

The items CUD have been implemented. They are all only granted to Seller and Admin Role. Please login with Seller or Admin user in order to access the CRUD. Please use below credentials:
1. jane@moka.com / 111111 (Seller Role)
2. nazmy@moka.com / password (Admin Role)

I intentionally created different roles and assigning different privileges to demonstrate the usage of cancancan.

The registration will only create user with regular role. An additional Business's user management shall be introduced to allow Admin manage Business's user. But I suppose, it's not necessary for this test. For now, we may change the role in the DB to assign admin role (role_id=3). 

crud items

Hope it clarifies.

Thanks,
Nazmy

**Comment by glaksmono** [comment]
Ah, perfect! 👍

**Pull request closed without merging**